zebra/dplane_fpm_nl.c

```diff
@@ -866,6 +884,70 @@ static int fpm_nl_enqueue(struct fpm_nl_ctx *fnc, struct zebra_dplane_ctx *ctx)
 	return 0;
 }
 
+/*
+ * LSP walk/send functions
+ */
+struct fpm_lsp_arg {
+	struct zebra_dplane_ctx *ctx;
+	struct fpm_nl_ctx *fnc;
+	bool complete;
+};
+
+static int fpm_lsp_send_cb(struct hash_bucket *bucket, void *arg)
+{
+	zebra_lsp_t *lsp = bucket->data;
+	struct fpm_lsp_arg *fla = arg;
+
+	/* Skip entries which have already been sent */
+	if (CHECK_FLAG(lsp->flags, LSP_FLAG_FPM))
+		return HASHWALK_CONTINUE;
+
+	dplane_ctx_reset(fla->ctx);
+	dplane_ctx_lsp_init(fla->ctx, DPLANE_OP_LSP_INSTALL, lsp);
+
+	if (fpm_nl_enqueue(fla->fnc, fla->ctx) == -1) {
+		fla->complete = false;
+		return HASHWALK_ABORT;
+	}
+
+	/* Mark entry as sent */
+	SET_FLAG(lsp->flags, LSP_FLAG_FPM);
+	return HASHWALK_CONTINUE;
+}
+
+static int fpm_lsp_send(struct thread *t)
+{
+	struct fpm_nl_ctx *fnc = THREAD_ARG(t);
+	struct zebra_vrf *zvrf = vrf_info_lookup(VRF_DEFAULT);
+	struct fpm_lsp_arg fla;
+
+	fla.fnc = fnc;
+	fla.ctx = dplane_ctx_alloc();
+	fla.complete = true;
+
+	hash_walk(zvrf->lsp_table, fpm_lsp_send_cb, &fla);
+
+	dplane_ctx_fini(&fla.ctx);
+
+	if (fla.complete) {
+		WALK_FINISH(fnc, FNE_LSP_FINISHED);
+
+		/* Now move onto routes */
+		if (fnc->use_nhg)
```

Comment:
Please, just add timer for `fpm_nhg_reset` instead of deciding nhg/rib reset.

The reasoning is simple: I found no benefit from queueing multiple work batches, but disadvantages when debuging FPM connection on startup (e.g. things would happen outside the expected order and testing by comparing output is harder).

I simplified it in this PR commit: https://github.com/FRRouting/frr/pull/7472/commits/18a58fd670612b7336cbce7d366a7ec9ac710cda

Also note that I delayed the `reset` to after we connected:
https://github.com/FRRouting/frr/pull/7472/commits/3214e90c2dc68a7c81a8ad8035367018e7d339b4

Reply:
Thanks for the review @rzalamena! Your PR changes and your suggestion sound good to me.

This PR is ready from my POV. I think if this one goes first then we should leave things as is (otherwise the `use_nhg` check will be entirely missing, until your changes come in)?

I'm happy to wait and fix things up after your PR is merged. Please let me know what you would prefer.